packages/cli/openapi-parser/src/schema/SchemaParserContext.ts

```diff
@@ -6,11 +7,20 @@ export interface SchemaParserContext {
     logger: Logger;
     DUMMY: SchemaParserContext;
 
+    sdkLanguage: generatorsYml.GenerationLanguage | undefined;
     shouldUseTitleAsName: boolean;
 
     referenceExists(ref: string): boolean;
     resolveSchemaReference(schema: OpenAPIV3.ReferenceObject): OpenAPIV3.SchemaObject;
     markSchemaAsReferencedByNonRequest(schemaId: SchemaId): void;
     markSchemaAsReferencedByRequest(schemaId: SchemaId): void;
     markReferencedByDiscriminatedUnion(schema: OpenAPIV3.ReferenceObject, discrminant: string, times: number): void;
+
+    shouldUseUndiscriminatedUnionsForDiscriminated: boolean;
+    getShouldUseUndiscriminatedUnionsForDiscriminated(): boolean;
```

Comment:
nit: why do we need both of these functions and not just one?

Reply:
ones the flag and the other is a function which evals if it should beused (really just checks the flag as well as the language)